### PR TITLE
Add command to change room and floor settings

### PR DESCRIPTION
**Pull request by teekoksiang**
Part of #180. 
Notes: The room and floor settings changes will take effect only after restarting the application. It might erase the existing data if the resident's room and floor number are no longer valid after updating the settings. It will be part of the advanced features.

### CODE REVIEW

File: src/main/java/seedu/address/model/ReadOnlyUserPrefs.java [hunk 1]
```diff
@@ -13,8 +13,12 @@
     Path getAddressBookFilePath();
     String[] getBlockSettings();
     int getMinRoomSettings();
+    void setMinRoomSettings(int minRoomSettings);
     int getMaxRoomSettings();
+    void setMaxRoomSettings(int maxRoomSettings);
     int getMinFloorSettings();
+    void setMinFloorSettings(int minFloorSettings);
     int getMaxFloorSettings();
+    void setMaxFloorSettings(int maxFloorSettings);
```
Comment (schoolex) [inline src/main/java/seedu/address/model/ReadOnlyUserPrefs.java]: i think set functions violates the `ReadOnly` function of this class. Can consider using the `UserPrefs' class instead

File: src/main/java/seedu/address/logic/commands/EditRoomCommand.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        }
+        if (maxRoomSettings < UserPrefs.MIN_ALLOWED_ROOMS || maxRoomSettings > UserPrefs.MAX_ALLOWED_ROOMS) {
+            throw new CommandException(String.format(MESSAGE_INVALID_MAX_ROOM,
+                UserPrefs.MIN_ALLOWED_ROOMS, UserPrefs.MAX_ALLOWED_ROOMS));
+        }
+        if (minRoomSettings > maxRoomSettings) {
+            throw new CommandException(MESSAGE_INVALID_ROOM_RANGE);
+        }
+
+        ReadOnlyUserPrefs userPrefs = model.getUserPrefs();
```
Comment (schoolex) [inline src/main/java/seedu/address/logic/commands/EditRoomCommand.java]: Can consider using `UserPrefs` instead

File: src/main/java/seedu/address/model/ReadOnlyUserPrefs.java (hunk 1 above)
Comment (teekoksiang) [reply src/main/java/seedu/address/model/ReadOnlyUserPrefs.java]: Good point, fixed.

**Review by schoolex — APPROVED** [review]
LGTM

**Pull request merged**